staging/src/k8s.io/apiserver/pkg/util/flowcontrol/apf_filter.go

```diff
@@ -55,7 +57,8 @@ type Interface interface {
 	// ctx is cancelled or times out.
 	Handle(ctx context.Context,
 		requestDigest RequestDigest,
-		workEstimator func(fs *flowcontrol.FlowSchema, pl *flowcontrol.PriorityLevelConfiguration, flowDistinguisher string) fcrequest.WorkEstimate,
+		noteFn func(fs *flowcontrol.FlowSchema, pl *flowcontrol.PriorityLevelConfiguration, flowDistinguisher string),
+		workEstimator func(flowSchemaName, priorityLevelName string) fcrequest.WorkEstimate,
```

Comment:
I recommend removing the parameters from the `workEstimator` so that there is no possibility of inconsistency between the values given to `workEstimator` and those given to `noteFn`.  Also, explicitly document that `workEstimator` is called, if at all, _after_ `noteFn`.